# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.cpp
# Commit: 07c8392cb04d
# Full Hash: 07c8392cb04d095caa8e65ee1a10ccc5f7352eb6
# Author: Andrew McCreight <continuation@gmail.com>
# Date: 2021-05-07 21:46:25
# Regressor Bug: 1618547
# File Overlap Count: 1
# Description:
#   Bug 1618547 - Support Fission in the process priority manager. r=kmag,gsvelto
#   
#   With Fission, there can be multiple BrowserParents in a single tab, so
#   this patch moves the tracking of active tabs onto the top BrowsingContext
#   in a tab. If the priority of a top BC is changed, then the activity
# ==============================================================================

diff -r 285861b32934 -r 07c8392cb04d dom/ipc/BrowserParent.cpp
--- a/dom/ipc/BrowserParent.cpp	Fri May 07 17:42:21 2021 +0000
+++ b/dom/ipc/BrowserParent.cpp	Fri May 07 17:51:34 2021 +0000
@@ -224,7 +224,6 @@
       mRemoteTargetSetsCursor(false),
       mPreserveLayers(false),
       mRenderLayers(true),
-      mActiveInPriorityManager(false),
       mHasLayers(false),
       mHasPresented(false),
       mIsReadyToHandleInputEvents(false),
@@ -234,6 +233,14 @@
   // When the input event queue is disabled, we don't need to handle the case
   // that some input events are dispatched before PBrowserConstructor.
   mIsReadyToHandleInputEvents = !ContentParent::IsInputEventQueueSupported();
+
+  // If we're in a BC tree that is active with respect to the priority manager,
+  // ensure that this new BrowserParent is marked as active. This ensures that
+  // the process will be prioritized in a cross-site iframe navigation in an
+  // active tab.
+  if (aBrowsingContext->Top()->IsPriorityActive()) {
+    ProcessPriorityManager::ActivityChanged(this, true);
+  }
 }
 
 BrowserParent::~BrowserParent() = default;
@@ -3377,13 +3384,6 @@
 bool BrowserParent::GetRenderLayers() { return mRenderLayers; }
 
 void BrowserParent::SetRenderLayers(bool aEnabled) {
-  if (mActiveInPriorityManager != aEnabled) {
-    mActiveInPriorityManager = aEnabled;
-    // Let's inform the priority manager. This operation can end up with the
-    // changing of the process priority.
-    ProcessPriorityManager::ActivityChanged(this, aEnabled);
-  }
-
   if (aEnabled == mRenderLayers) {
     if (aEnabled && mHasLayers && mPreserveLayers) {
       // RenderLayers might be called when we've been preserving layers,
@@ -3446,13 +3446,6 @@
   }
 }
 
-void BrowserParent::Deprioritize() {
-  if (mActiveInPriorityManager) {
-    ProcessPriorityManager::ActivityChanged(this, false);
-    mActiveInPriorityManager = false;
-  }
-}
-
 bool BrowserParent::StartApzAutoscroll(float aAnchorX, float aAnchorY,
                                        nsViewID aScrollId,
                                        uint32_t aPresShellId) {